save-backend/src/main/kotlin/com/saveourtool/save/backend/security/ProjectPermissionEvaluator.kt

```diff
@@ -88,11 +86,15 @@ class ProjectPermissionEvaluator {
 
     private fun Authentication.hasRole(role: Role): Boolean = authorities.any { it.authority == role.asSpringSecurityRole() }
 
-    private fun hasWriteAccess(userId: Long?, project: Project, projectRole: Role): Boolean = when {
-        userId == null -> false
-        project.userId == userId -> true
-        else -> projectRole == Role.ADMIN || projectRole == Role.OWNER
-    }
+    private fun hasReadAccess(userId: Long?, projectRole: Role): Boolean = hasWriteAccess(userId, projectRole) ||
```

Comment:
Public projects should also grant read access for everyone, right?

Reply:
Well, the name of the method is quite confusing. `has...Access` means having a special role while public project check is done explicitly in `hasPermission` method